### PR TITLE
Add some assertions to sanity check the tech plugin

### PR DESCRIPTION
**Pull request by jwright6323**
We ran into issues with a malformed tech plugin that silently generated nonsensical power straps. Also funnily enough, our place and route tool of choice does NOT error out when you specify a negative power strap width...

### CODE REVIEW

File: src/hammer-tech/stackup.py [hunk 1]
```diff
@@ -279,6 +279,7 @@ def get_width_spacing_start_twwt(self, tracks: int, force_even: bool = False) ->
         """
         widths_and_spacings = self.power_strap_widths_and_spacings
         spacing = widths_and_spacings[0].min_spacing
+        assert self.pitch - self.min_width == spacing, "Tech plugin is malformed for metal {}, the minimum spacing in the width-spacing list must be the same as (pitch - min_width).".format(self.name)
```
Comment (edwardcwang) [inline src/hammer-tech/stackup.py]: Minor point, can we add unit tests for these kinds of bad cases?

Comment (jwright6323) [reply src/hammer-tech/stackup.py]: IDK what you mean- there are no unit tests in the tech plugin.

Comment (edwardcwang) [reply src/hammer-tech/stackup.py]: Well right now they live in `hammer-vlsi/tech_test.py` IIRC

Comment (jwright6323) [reply src/hammer-tech/stackup.py]: but this isn't a problem with the tech code- it's catching a problem with a bad plugin. Are you saying you want a test that contains a bad plugin?

Comment (edwardcwang) [reply src/hammer-tech/stackup.py]: Yeah, a test that purposely feeds it a bad plugin and check that it's caught

Comment (jwright6323) [reply src/hammer-tech/stackup.py]: I think that's beyond the scope of this PR, but we can file an issue.

Comment (jwright6323) [reply src/hammer-tech/stackup.py]: Filed #402

**Pull request merged**